### DISCUSSION TITLE
Inheritance question

### DISCUSSION THREAD

**Discussion by mmintel** [prisma/prisma#8242] [Q&A] [answered]
Hi there, currently getting started with prisma as a frontend developer, so my knowledge about database design is quite limited. Before I run into the wrong direction I have a question about this here. I got a Profile that contains a Feed with different types of items e.g. Posts and Repos. Each have unique properties. I have modeled it like this for now:

```prisma
model Post {
  id          String       @id @default(cuid())
  title       String
  description String
  likes       Int
  comments    Int
  provider    PostProvider
  feed        Feed         @relation(fields: [feedId], references: [id])
  feedId      String
}

model Repo {
  id       String       @id @default(cuid())
  title    String
  stars    Int
  forks    Int
  provider RepoProvider
  feed     Feed         @relation(fields: [feedId], references: [id])
  feedId   String
}

model Feed {
  id        String   @id @default(cuid())
  visible   Boolean  @default(true)
  createdAt DateTime @default(now())
  url       String
  Post      Post[]
  Repo      Repo[]
  Profile   Profile?
}

model Profile {
  id        String   @id @default(cuid())
  published Boolean  @default(false)
  feed      Feed     @relation(fields: [feedId], references: [id])
  owner     User?    @relation(fields: [ownerId], references: [id])
  ownerId   String?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  feedId    String
}
```

This way I have to list each post type in the feed as
```
  Post      Post[]
  Repo      Repo[]
```

which fields quite redundant. Is it also possible to design it so that I just define a single field to represent either Post or Repo as one feed item can never be both at the same time.

Cheers!

**Comment by ryands17** [comment, marked as answer]
@mmintel 👋 

Unfortunately this isn't possible directly with Prisma, but we do have a feature request for the same here so it would be great if you could add a 👍 to this so that we can know the priority for this 🙂 

https://github.com/prisma/prisma/issues/2505


As a workaround, this type of inheritance can be designed in your application logic if you're using GraphQL. Here's a great resource on how to do that: https://graphql.org/learn/schema/#union-types

Let me know if you have any other questions.

**Reply by mmintel** [reply]
Thanks for your quick response! Now I can at least stop thinking about it :)

**Reply by sandbox-apps** [reply]
Just to follow while I'm developing my project and not production.. What will be my goto approach for this?
Should I use single table for all inheritance because that what you plan or supertype subtype approach?
At least for I know what structure is needed before the feature is coming so that if the feature is implemented I dont have to self destruct myself because of the schema/architecture I implemented is not compatible with prisma union...

@ryands17

**Reply by ryands17** [reply]
@sandbox-apps I wouldn't rely on Prisma implementing unions yet as that's not currently on the [roadmap](https://www.notion.so/prismaio/Prisma-public-roadmap-50766227b779464ab98899accb98295f) so it would be best to structure this as separate model (subtype) approach.

**Reply by sandbox-apps** [reply]
Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
